apiVersion: openapi/v1alpha1
kind: Radar
metadata:
    name: "19097334"
    labels:
        datastore_id: "5874835112591360"
data:
    classification: Crash/Hang/Data Loss
    created: "2014-11-30T20:06:19.801794Z"
    description: |-
        Summary:
        A generic function nested within an outer function called from a sibling function crashes.

        This seemed really familiar, but it isn’t in the release notes for Xcode 6.2 and I don’t have the ones for 6.1.1 handy. (It crashes with Xcode 6.2’s swift anyway.)


        Steps to Reproduce:
        1. Compile this code:
        func f() {
        	func g<T>(h: T -> ()) -> () {
        		return ()
        	}
        	func i() {
        		g { (x: Int) in () }
        	}
        }


        Expected Results:
        Not crash.


        Actual Results:
        Crash.


        Regression:
        Moving g() up out of f() resolves the crash.
        Seems to depend on g being generic.


        Notes:
        Crash log attached.
    email: rix.rob@gmail.com
    modified: "2014-11-30T20:06:19.801934Z"
    number: "19097334"
    number_intvalue: 19097334
    originated: 30-Nov-2014 03:06 PM
    parent_number: '&{NULL_VALUE}'
    product: Developer Tools
    product_version: Xcode 6.1.1 (6A2006)
    reproducible: Always
    resolved: ""
    status: Open
    title: 'Swift: Calling nested generic functions from nested sibling functions crashes the compiler'
